### ISSUE TITLE
How to make a server ?

### ISSUE THREAD

**Issue by gamerulady** [qwugfd/mempsrc#4]
I dont know how to do a server on mirrors edge pls help me. I need hamachi or something?

**Comment by qwugfd** [comment]
hamachi should work, or openvpn or any vpn. sometimes such a vpn service even comes with your router.
for example the avm fritzbox here in germany has its own vpn service.
but it also should work without vpn. just do the port forwarding and make sure the windows firewall or an antivirus doesnt block it.
if you download some strange exe from the internet that tries to contact strange ips, antiviruses simply block it without even notifying you.
then just whitelist the server.exe and the multiplayer.exe.
the steps are the same as with any other game where you can set up a server, e.g. minecraft.
there should be lots of tutorials on how to set up a minecraft server.
if you follow these and can set up a minecraft server, you should know enough to set up the mirrors edge server.
if it still doesnt work, please let me know exactly what happened. be very specific.
e.g. "i started server.exe then mirrorsedge.exe, then it crashed after the splash screen but before the title screen."
something like that i can work with, i cant work with general stuff.

http://www.noip.com/support/knowledgebase/how-to-host-and-configure-a-minecraft-server-with-no-ip/

http://www.noip.com/support/knowledgebase/getting-started-with-no-ip-com/

http://www.noip.com/blog/2011/05/19/got-game-how-to-host-your-own-game-server-with-no-ip/

http://www.noip.com/support/knowledgebase/basic-troubleshooting-guide/

http://minecraft.gamepedia.com/Tutorials/Setting_up_a_server

if you use noip, in the multiplayersettings.ini instead of the ip simply write the name,
so instead of e.g 88.120.45.32 you write mymemultiplayeradress1337420formeandmybff.ddns.net, thats all.
then you dont have to worry about changing ip adresses ever again.

**Comment by gamerulady** [comment]
You can do a video (tutorial) pls ?

**Comment by qwugfd** [comment]
no. i would have to edit out all my private stuff that shows up in such a video, like private phone numbers, etc. and then for your router it would look different anyway (unless you happen to have a fritzbox too). when it comes to video editing im the laziest person on the planet. but maybe if you follow everything i said above, read everything and then know everything, you can make a tutorial? pls?

**Comment by gamelaster** [comment]
@qwugfd lol, nice to see you active :D where did you get that reverse engineering experiences?

**Comment by gamerulady** [comment]
My friend do the server like you seyd but i cant se him or me.And somewere its Celest just stay and do nothing pls help me.

**Comment by qwugfd** [comment]
well "celeste" and "standing there" tells me two things: 1. you were playing with three people, cause with two people celeste shouldnt be showing initially, only kate. 2. it got the initial packet to show celeste in the level, but then didnt get further packets for her animation. keep in mind that a lot of data is sent. if you have 1 mbit upload, youre on skype with your friends, and maybe even streaming to twitch, your upload limit is reached and the packets from the multiplayer dont get through. make sure you have enough upload left. and then if this happens, try the following: everyone reload their level. so if everyone is in the prologue, everyone restart the prologue. if this doesnt help, im afraid i cant help you, cause it means that the memory loactions of the meshes change on your system in a way i havent experienced on my system and cant replicate here.

**Comment by qwugfd** [comment]
@gamelaster i dont seem to know you. maybe i accidentally took someone elses name?

**Comment by qwugfd** [comment]
you can also play without the detailed animations: everyone press L. then the memory locations of the bones shouldnt be a problem, since they arent used. when i get a new computer i will make sure the memp runs on it. this can take a while though, since my 6 year old one doesnt show any signs of giving up and i can do everything i want on it.
